### PR TITLE
Tests for absolute path in exceptions

### PR DESCRIPTION
**Pull request by danmoseley**
Tests for https://github.com/dotnet/coreclr/pull/27470

### CODE REVIEW

File: src/System.Reflection/tests/AssemblyTests.cs [hunk 1]
```diff
@@ -318,9 +318,18 @@ public void LoadFile_NullPath_Netcore_ThrowsArgumentNullException()
         }
 
         [Fact]
-        public void LoadFile_NoSuchPath_ThrowsArgumentException()
+        public void LoadFile_NoSuchPath_ThrowsFileNotFoundException()
         {
-            AssertExtensions.Throws<ArgumentException>("path", null, () => Assembly.LoadFile("System.Runtime.Tests.dll"));
+            string path = @"c:\foo\bar";
```
Comment (stephentoub) [inline src/System.Reflection/tests/AssemblyTests.cs]:
```suggestion
            string path = Guid.NewGuid().ToString("N");
```

File: src/System.Reflection/tests/AssemblyTests.cs [hunk 2]
```diff
@@ -318,9 +318,18 @@ public void LoadFile_NullPath_Netcore_ThrowsArgumentNullException()
...
+            AssertExtensions.ThrowsContains<FileNotFoundException>(() => Assembly.LoadFile(path), path);
+        }
+
+        [Fact]
+        public void LoadFile_PartiallyQualifiedPath_ThrowsArgumentException()
+        {
+            string path = "System.Runtime.Tests.dll";
+            ArgumentException ex = AssertExtensions.Throws<ArgumentException>("path", () => Assembly.LoadFile(path));
+            Assert.Contains(path, ex.Message);
```
Comment (stephentoub) [inline src/System.Reflection/tests/AssemblyTests.cs]: Why add a ThrowsContains for one pattern but not for this one?

File: src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs [hunk 3]
```diff
@@ -44,6 +44,30 @@ public static void GetAssemblyNameTest_NullParameter()
             Assert.Throws<ArgumentNullException>(() => AssemblyLoadContext.GetAssemblyName(null));
         }
 
+        [Fact]
+        public static void LoadFromAssemblyPath_PartiallyQualifiedPath_ThrowsArgumentException()
+        {
+            string path = @"foo\bar.dll";
```
Comment (stephentoub) [inline src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs]:
```suggestion
            string path = Path.Combine("foo", "bar.dll");
```

File: src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs [hunk 4]
```diff
@@ -44,6 +44,30 @@ public static void GetAssemblyNameTest_NullParameter()
...
+            ArgumentException ex = AssertExtensions.Throws<ArgumentException>("assemblyPath", () => (new AssemblyLoadContext("alc")).LoadFromAssemblyPath(path));
+            Assert.Contains(path, ex.Message);
+        }
+
+        [Fact]
+        public static void LoadFromNativeImagePath_PartiallyQualifiedPath_ThrowsArgumentException()
+        {
+            string path = @"foo\bar.dll";
```
Comment (stephentoub) [inline src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs]:
```suggestion
            string path = Path.Combine("foo", "bar.dll");
```

File: src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs [hunk 5]
```diff
@@ -44,6 +44,30 @@ public static void GetAssemblyNameTest_NullParameter()
...
+            ArgumentException ex = AssertExtensions.Throws<ArgumentException>("nativeImagePath", () => (new AssemblyLoadContext("alc")).LoadFromNativeImagePath(path, null));
+            Assert.Contains(path, ex.Message);
+        }
+
+        [Fact]
+        public static void LoadFromNativeImagePath_PartiallyQualifiedPath_ThrowsArgumentException2()
+        {
+            string path = @"foo\bar.dll";
```
Comment (stephentoub) [inline src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs]:
```suggestion
            string path = Path.Combine("foo", "bar.dll");
```

File: src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs [hunk 6]
```diff
@@ -44,6 +44,30 @@ public static void GetAssemblyNameTest_NullParameter()
...
+            ArgumentException ex = AssertExtensions.Throws<ArgumentException>("assemblyPath", () => (new AssemblyLoadContext("alc")).LoadFromNativeImagePath(@"c:\baz\boz", path));
```
Comment (stephentoub) [inline src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs]: Same comment about this path

**Review by stephentoub — APPROVED** [review]
Other than the comments, looks fine. Thanks.

File: src/Common/tests/CoreFx.Private.TestUtilities/System/AssertExtensions.cs [hunk 7]
```diff
@@ -21,6 +21,12 @@ public static void Throws<T>(Action action, string expectedMessage)
             Assert.Equal(expectedMessage, Assert.Throws<T>(action).Message);
         }
 
+        public static void ThrowsContains<T>(Action action, string expectedMessageContent)
+            where T : Exception
+        {
+            Assert.Contains(expectedMessageContent, Assert.Throws<T>(action).Message);
```
Comment (bartonjs) [inline src/Common/tests/CoreFx.Private.TestUtilities/System/AssertExtensions.cs]: I've generally been guarding any exception message tests with the culture being en-US (though many of mine come from Win32 FormatMessage).  Should this have a culture guard in it, or indicate that it's only applicable to all-culture message testing? (e.g. the bad parameter value is in the message)

**Comment by danmosemsft** [comment]
https://github.com/dotnet/corefx/issues/42211

File: src/System.Reflection/tests/AssemblyTests.cs (hunk 2 above)
Comment (danmosemsft) [reply src/System.Reflection/tests/AssemblyTests.cs]: contains + argument seemed more obscure case.

File: src/Common/tests/CoreFx.Private.TestUtilities/System/AssertExtensions.cs (hunk 7 above)
Comment (danmosemsft) [reply src/Common/tests/CoreFx.Private.TestUtilities/System/AssertExtensions.cs]: Wouldn't the parameter value be found within the message in every culture?

File: src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs (hunk 6 above)
Comment (danmosemsft) [reply src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs]: OK, `\foo` is qualified on all OS...

Comment (stephentoub) [reply src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs]: > `\foo` is qualified on all OS

What do you mean?

**Pull request merged**